lib/actions/post.actions.ts

```diff
@@ -37,3 +34,153 @@ export async function createPost(data: IPostSchema) {
   }
   return { error: "An unexpected error occurred while creating post." };
 }
+
+export async function getAllPosts({ page }: { page: string }) {
+  const postsToTake = 4;
+  let hasNextPage = false;
+  try {
+    const userEmail = await getUserSession();
+    const somePosts = await prisma.post.findMany({
+      where: {
+        userEmail,
+      },
+      orderBy: [{ createdAt: "desc" }],
+      skip: (Number(page) - 1) * postsToTake,
+      take: postsToTake + 1,
+    });
+
+    if (somePosts.length > postsToTake) {
+      somePosts.pop();
+      hasNextPage = true;
+    }
+
+    const allPosts = await prisma.post.findMany({
+      where: {
+        userEmail,
+      },
+    });
+
+    const numberOfPages = Math.ceil(allPosts.length / postsToTake);
+
+    return { somePosts, hasNextPage, numberOfPages };
```

Comment:
```suggestion
    const { _count: numberOfResults } = await prisma.post.aggregate({
      _count: true,
      where: {
        userEmail,
        ...(searchTerm &&
          searchTerm !== "all" && { createType: searchTerm as CreateType }),
      },
    });

    return {
      somePosts,
      hasNextPage,
      numberOfPages: Math.ceil(numberOfResults / 4) || 1,
    };
```

We need to also do the filtering here to get the correct count. We can use 'aggregate' instead of just counting the results to be a little more efficient as well. 

Aggregate will return the number of results to us, instead of all of the results (this will save our server from sending all the posts and all the information back and forth to the database.)

https://www.prisma.io/docs/orm/prisma-client/queries/aggregation-grouping-summarizing#aggregate

Reply:
The `|| 1` part I added is just to make sure we show 1/1 instead of 1/0 if there are no results :)